tables/tests/test_carray.py

```diff
@@ -2209,7 +2209,7 @@ def test00b_zeros(self):
             print("First row-->", ca[0])
             print("Defaults-->", ca.atom.dflt)
         self.assertTrue(allequal(ca[0], numpy.zeros(N, 'S3')))
-        self.assertTrue(allequal(ca.atom.dflt, numpy.zeros(N, 'S3')))
+        self.assertTrue(allequal(ca.atom.dflt, b""))
```

Comment:
What do you mean by 'multidimensional dtypes'?  

`h5py` should support all dtypes that hdf5 supports (and if any are missing it should be fixed in h5py).